### ISSUE TITLE
Airdrop tool does not highlight erroneous accounts/invalid UIDs.

### ISSUE THREAD

**Issue by bluabaleno** [Novusphere/discussions-app#187]
A basic check for whether the EOS account name fits the convention of:

EOS Account names must be exactly 12 characters long and consist of lower case characters and digits up until 5.

and whether or not the UID conforms to the EOS public key format should resolves most cases. 

There is no need to actually look up whether or not each account/UID actually exists.

**Comment by Novusphere** [comment]
@bluabaleno this does not work for EOS accounts with "domains", i.e. I can register "xia.x" easily, which would not conform to the standards you laid out.

**Comment by bluabaleno** [comment]
@Novusphere ah right, I forgot about the domain part. 

> The transaction will fail if a single account is invalid.

Right, but the one doing the airdrop will have no idea which is invalid. At which point they would have to search individual which one is valid or not. 

Or do you mean to say that in cases where the account name is valid, but does not exist?

i.e bigbluephale is a valid name, but does not exist.

In that case, there is very little we can do but to say that one of the account might not exist or else we would have to perform a check ourselves. Which in list with hundreds of accounts might take a while?

**Comment by Novusphere** [comment]
> i.e bigbluephale is a valid name, but does not exist.

> In that case, there is very little we can do but to say that one of the account might not exist or else we would have to perform a check ourselves. Which in list with hundreds of accounts might take a while?

The way the airdrop tool currently works, it validates all the names you put and ensures that they are *real* EOS accounts that **exist**. If they don't exist, it reports it to you. Perhaps it would be preferred that if they don't exist, it ignores them and just produces a list of accounts for which the airdrop failed because they were invalid?

**Comment by bluabaleno** [comment]
Oh, I didn't know that was the case. I assumed there was no check at all. 

ok then it is working as intended. And I assume for UIDs, as long as they are valid, even if someone haven't registered for an UID with it, it still can be dropped to? (Unlikely but just want to make sure)

I tested the tool again and got the erroneous account name warning this time but it only lists one when there were more than one that exists in the list.

**Comment by bluabaleno** [comment]
Also, highlighting the invalid account name would make it much easier to remove the name from the list

**Comment by Novusphere** [comment]
> ok then it is working as intended. And I assume for UIDs, as long as they are valid, even if someone haven't registered for an UID with it, it still can be dropped to? (Unlikely but just want to make sure)

> I tested the tool again and got the erroneous account name warning this time but it only lists one when there were more than one that exists in the list.

@bluabaleno Yes, as long as it's a valid public key (UID) it will airdrop to it. It doesn't check if it exists. We could do this. For it to "exist" the user will have needed to have posted at least once. 

Regarding the errors you mentioned, I believe it only catches one error at a time. As in, if it detects an invalid account it reports that it's invalid and doesn't bother checking the rest, where in there may be several that are invalid. I'll have Gux change this so that it detects all that are invalid in one shot.
